src/server/route/routes/assignment.ts

```diff
@@ -0,0 +1,153 @@
+import { router } from '../router';
+import { Request, Response } from 'express';
+import { db } from '../../dbConfigs';
+import { checkProjectPermission } from '../../helper/permissionHelper';
+
+const assignmentCheck = async (
+    req: Request,
+    res: Response
+): Promise<[number, number, number] | undefined> => {
+    const userId: number = parseInt(req.params.userId);
+    const nodeId: number = parseInt(req.params.nodeId);
+
+    if (!userId || !nodeId) {
+        res.status(403).json({ message: 'Invalid user assignment' });
+        return;
+    }
+
+    //check user id
+
+    const userQuery = await db.query(
+        'SELECT COUNT(*) FROM users WHERE id = $1;',
+        [userId]
+    );
+
+    if (!Number(userQuery.rows[0].count)) {
+        res.status(403).json({ message: 'Invalid user id' });
+        return;
+    }
+
+    //get project id
+
+    const projectIdQuery = await db.query(
+        'SELECT project_id FROM node WHERE id = $1;',
+        [nodeId]
+    );
+
+    if (!projectIdQuery.rowCount) {
+        res.status(403).json({ message: 'Invalid node id' });
+        return;
+    }
+
+    const projectId: number = projectIdQuery.rows[0].project_id;
+
+    //check permissions
+
+    const permissions = await checkProjectPermission(req, projectId);
+
+    if (!permissions.edit) {
+        res.status(401).json({ message: 'No permission' });
+        return;
+    }
+
+    //return user, node and project ids
+
+    return [userId, nodeId, projectId];
+};
+
+router
+    .route('/assignment/assign/:nodeId/:userId')
+    .post(async (req: Request, res: Response) => {
+        const result = await assignmentCheck(req, res);
+        if (!result) return;
+
+        const [userId, nodeId, projectId] = result;
+
+        //check whether user belongs to project
+
+        const projectBelongQuery = await db.query(
+            'SELECT COUNT(*) FROM userBelongProject WHERE project_id = $1 AND users_id = $2;',
```

Comment:
Small heads up but userBelongProject has been renamed to users__project, so you probably want to merge the branch "invite-users-frontend" again and change this table name.

Reply:
LGTM after you do this

Reply:
done